src/Http/Http/src/StreamPipeReader.cs

```diff
@@ -68,12 +71,25 @@ public StreamPipeReader(Stream readingStream) : this(readingStream, minimumSegme
         /// </summary>
         /// <param name="readingStream">The stream to read from.</param>
         /// <param name="minimumSegmentSize">The minimum segment size to return from ReadAsync.</param>
-        /// <param name="pool"></param>
-        public StreamPipeReader(Stream readingStream, int minimumSegmentSize, MemoryPool<byte> pool = null)
+        /// <param name="pool">The MemoryPool to use</param>
+        public StreamPipeReader(Stream readingStream, int minimumSegmentSize, MemoryPool<byte> pool)
+            : this(readingStream, minimumSegmentSize, minimumReadThreshold: 0, pool)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new StreamPipeReader.
+        /// </summary>
+        /// <param name="readingStream">The stream to read from.</param>
+        /// <param name="minimumSegmentSize">The minimum segment size to return from ReadAsync.</param>
+        /// <param name="minimumReadThreshold">The threshold for obtaining a new block of memory to call ReadAsync on.</param>
+        /// <param name="pool">The MemoryPool to use</param>
+        public StreamPipeReader(Stream readingStream, int minimumSegmentSize, int minimumReadThreshold, MemoryPool<byte> pool)
```

Comment:
4 constructors already? I think we need to make an options type.

Reply:
Sure.